doc/src/base/collections.md

```diff
@@ -235,6 +236,61 @@ Partially implemented by:
   * [`Iterators.Pairs`](@ref)
 
 ## Set-Like Collections
+A [`Set`](@ref) is a collection of elements with no duplicated elements and keys/indices (unlike `Array` or `Dict`). So in a `Set` you can have only one of each element, and the order of elements isn't important.
+
+```jldoctest set_collections
+julia> colors = Set(["yellow", "blue", "green", "red"])
+Set{String} with 4 elements:
+  "yellow"
+  "blue"
+  "green"
+  "red"
+```
+
+Use [`push!`](@ref) to add elements to a set:
+```jldoctest set_collections
+julia> push!(colors, "black")
+Set{String} with 5 elements:
+  "yellow"
+  "blue"
+  "green"
+  "black"
+  "red"
+```
+
+Since sets don't store repeated elements, `push!` will not have any effect for pushing the elements that are already in the set. Also because `Set` doesn't have a concept of "first" (unlike `Array`), using [`pushfirst!`](@ref) to insert an item at the beginning of a set will result in an error.
```

Comment:
I feel like this whole paragraph can be removed. This is mostly rehashing things that have already been mentioned.

Reply:
Here I am describing the context of `push!`.